packages/plugin-npm/sources/NpmSemverResolver.ts

```diff
@@ -70,6 +70,17 @@ export class NpmSemverResolver implements Resolver {
     const manifest = new Manifest();
     manifest.load(registryData.versions[version]);
 
+    // Manually add node-gyp dependency if there is a script using it and not already set
+    const ident = structUtils.parseIdent(`node-gyp`);
```

Comment:
Use `makeIdent` instead - you also should move that in a constant at the top of the file to avoid recomputations (cf [PnpLinker](https://github.com/yarnpkg/berry/blob/master/packages/plugin-pnp/sources/PnpLinker.ts#L9) where I do something similar).